### ISSUE TITLE
Removed `SourceCode` methods create issues with ESLint 10

### ISSUE THREAD

**Issue by kevin-easygo** [sveltejs/eslint-plugin-svelte#1515]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

10.2.0

### What version of `eslint-plugin-svelte` are you using?

3.17.0

### What did you do?

<details>
<summary>Configuration</summary>

```ts
// eslint.config.ts

import tseslint from 'typescript-eslint';
import svelte from 'eslint-plugin-svelte';
import svelteParser from 'svelte-eslint-parser';
import { defineConfig } from 'eslint/config';

const config = {
  files: ['**/*.svelte', '**/*.svelte.ts'],
  languageOptions: {
    parser: svelteParser,
    parserOptions: {
      parser: tseslint.parser,
      extraFileExtensions: ['.svelte'],
    },
  },
};

export default defineConfig(
  svelte.configs['flat/recommended'],
  svelte.configs['flat/prettier'],
);
```
</details>

```svelte
<script lang="ts">
  $: test = () => {
    console.log("test");
  };
</script>
```

### What did you expect to happen?

ESLint should catch the `svelte/no-reactive-functions` error and report it.

### What actually happened?

ESLint throws `TypeError: source.isSpaceBetweenTokens is not a function` due to the deprecation of some `SourceCode` methods from ESLint 10 ([reference](https://eslint.org/docs/latest/use/migrate-to-10.0.0#-removal-of-deprecated-sourcecode-methods)).

### Link to **GitHub Repo** with Minimal Reproducible Example

https://eslint-online-playground.netlify.app/#eNplkeFKwzAUhV/lEvyxwZZuOhXqNgR9i0WhtLclM01Ck41J6bt702Rq9U8ber5zcs9tz1xXZngpWquQuzMqjyxnW1d20vq90AA3OXh0HnYwm8NuD334CFAa7Qx5lGlmggVCsPlT0Aaht1kKYAuGTkntOfG1bPjRUbxsrek89FBhLTW+jBIMUHemBZEcWXQIRqHJcHRX5DkxR/dbj/NPY5ZWnRqpl1EbaaHxMvJ0fXFS4/t7jFmocIgdoyeN7g6C1arwWYelaVvUFVaCvQWSHqE6lbVF+VE0SDWNpqLjrmg7n5ZuzunUmuqk6LyIQoXnV7QhSpcSqUt+Xe+0YrC+r1d8xdfJG4gI/Ki3fPVP/Vt/ZO/4+nHCTtR7vnngG8HSvxzY8AXZDbRf
